exporters/otlp/profiles/src/main/java/io/opentelemetry/exporter/otlp/internal/data/ImmutableAttributeUnitData.java

```diff
@@ -25,8 +25,8 @@ public abstract class ImmutableAttributeUnitData implements AttributeUnitData {
    *
    * @return a new AttributeUnitData mapping the given key to the given unit.
    */
-  public static AttributeUnitData create(long attributeKey, long unitIndex) {
-    return new AutoValue_ImmutableAttributeUnitData(attributeKey, unitIndex);
+  public static AttributeUnitData create(int attributeKeyStrindex, int unitStrindex) {
+    return new AutoValue_ImmutableAttributeUnitData(attributeKeyStrindex, unitStrindex);
```

Comment:
At the profiles.proto level, all fields referencing into the lookup tables now explicitly use an index or indices naming suffix for clarity. Most fields you can infer which table is being referenced from the name, e.g. function_index, but ones into the string table are used for a variety or purposes so the _strindex suffix makes clearer the lookup type.
At the Java level, directly mapping those field name does look a bit odd. I was toying with ...StringIndex instead of ...Strindex, or even just ...Index and rely on the Java type system to provide sufficient indication that it's a String. Any preference?

Reply:
I went looking for other examples of fields in `opentelemetry-proto` which utilize abbreviations, and what we call those in `opentelemetry-java`. I didn't find many, but found:

* [AnyValue.kvlist_value](https://github.com/open-telemetry/opentelemetry-proto/blob/2080d87c780b5a8c65c9ae2771130fd669cef480/opentelemetry/proto/common/v1/common.proto#L37) - called [KeyValue](https://github.com/open-telemetry/opentelemetry-java/blob/main/api/all/src/main/java/io/opentelemetry/api/common/KeyValue.java) in opentelemetry-java
* (arguably) [min and max](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/metrics/v1/metrics.proto#L597-L600) are abbreviations for minimum and maximum - we call these [min and max](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data/ExponentialHistogramPointData.java#L59-L75) in opentelemetry-java

So there is no consistent pattern, but not many examples to go on. The lack of examples may be an indication that profiling should adjust the proto field names to not be abbreviated. 

> I was toying with ...StringIndex instead of ...Strindex, or even just ...Index and rely on the Java type system to provide sufficient indication that it's a String. 

I have a slight preference for StringIndex, since Strindex isn't a word I've heard before so I had to do some deducing to figure out what it meant. Calling it Index and relying on the type system requires similar deducing. I realize I'm now falling into the classic java developer trope of preferring verbose names.. 😅

Reply:
> I have a slight preference for StringIndex, since Strindex isn't a word I've heard before

makes sense to me, there was a similar comment here: https://github.com/open-telemetry/opentelemetry-proto/pull/592#pullrequestreview-2408132184

Reply:
Changed to StringIndex